File: project6/readability.py
# Computes the Coleman-Liau index of text inputted by user

from cs50 import get_string

# Text input
text = get_string("Text: ")

# Count letters
def count_letter(text):
    count = 0
    for c in text:
        if (ord(c) >= 65 and ord(c) <= 90) or (ord(c) >= 97 and ord(c) <= 122):
            count += 1
    return count

# Count words
def count_word(text):
    count = 1
    for i in range(len(text) - 1):
        if ord(text[i]) == 32 and ((ord(text[i + 1]) >= 21 and ord(text[i + 1]) <= 126)):
            count += 1
    return count

# Count sentences
def count_sentence(text):
    count = 0
    for c in text:
        if ord(c) == 33 or ord(c) == 46 or ord(c) == 63:
            count += 1
    return count

# Calculate Coleman-Liau index
def index(text):
    num_letters = count_letter(text)
    print(num_letters)
    num_words = count_word(text)
    print(num_words)
    num_sentences = count_sentence(text)
    print(num_sentences)
    # Number words per 100 letters
    letters_word = num_letters / num_words * 100
    print(letters_word)
    # Number sentences per 100 letters
    sentences_word = num_sentences / num_words * 100
    print(sentences_word)
    # Calculate index, rounding to integer
    index = int(round(0.0588 * letters_word - 0.296 * sentences_word - 15.8))
    # Print grade level according to instructions
    if index < 1:
        print("Before Grade 1")
    elif index > 16:
        print("Grade 16+")
    else:
        print(f"Grade {index}")

# Print Coleman-Liau index
index(text)

File: pset2/readability/readability.c
// Computes the Coleman-Liau index of text inputted by user

#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <ctype.h>
#include <math.h>

int count_letters(string text);
int count_words(string text);
int count_sentences(string text);
int index(string text);

int main(void)
{
    // Get text from user and print
    string text = get_string("Text: ");
    // Print CL grade according to requested convention
    int grade = index(text);
    if (grade < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (grade >= 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", grade);
    }
}

// Count number of letters
int count_letters(string text)
{
    int num = 0;
    for (int i = 0; i < strlen(text); i++)
    {
        if (isupper(text[i]) != 0 || islower(text[i]) != 0)
        {
            num++;
        }
    }
    return num;
}

// Count number of words
int count_words(string text)
{
    int num = 0;
    for (int i = 0; i < strlen(text); i++)
    {
        if ((text[i] >= 21 && text[i] <= 126) && text[i + 1] == 32)
        {
            num++;
        }
    }
    num++;
    return num;
}

// Count number of sentences
int count_sentences(string text)
{
    int num = 0;
    for (int i = 0; i < strlen(text); i++)
    {
        if ((isupper(text[i]) != 0 || islower(text[i]) != 0) &&
            (text[i + 1] == 46 || text[i + 1] == 33 || text[i + 1] == 63))
        {
            num++;
        }
    }
    return num;
}

// Calculate Coleman-Liau index
int index(string text)
{
    int num_letters = count_letters(text);
    int num_words = count_words(text);
    int num_sentences = count_sentences(text);
    // Number words per 100 letters
    float letters_word = (float) num_letters / (float) num_words * 100;
    // Number sentences per 100 letters
    float sentences_word = (float) num_sentences / (float) num_words * 100;
    // Calculate index, rounding to integer
    float ind_float = round(0.0588 * (float) letters_word - 0.296 *
		(float) sentences_word - 15.8);
    // Convert index to integer
    int index = (int) ind_float;
    return index;
    printf("%i\n", index);
}
